core/objlist.hpp

```diff
@@ -281,6 +282,30 @@ class ObjList : public ObjListBase {
         del_bitmap_.swap(tmp_bool);
     }
 
+    size_t estimated_storage_size(const double sample_rate = 0.005) {
+        const size_t sample_num = this->get_size() * sample_rate;
+        BinStream bs;
+
+        // sample
+        std::unordered_set<size_t> sample_container;
+        std::random_device rd;
+        std::mt19937 generator(rd());
+        std::uniform_real_distribution<double> distribution(0.0, 1.0);
+        while (sample_container.size() < sample_num) {
+            size_t index = distribution(generator) * objlist_data_.get_vector_size();
+            if (!del_bitmap_[index])
+                sample_container.insert(index);
+        }
+
+        // log the size
+        for (auto iter = sample_container.begin(); iter != sample_container.end(); ++iter)
+            bs << objlist_data_.data_[*iter];
+
+        std::vector<ObjT>& v = objlist_data_.data_;
+        size_t ret = bs.size() * sizeof(char) * v.capacity() / sample_num;
```

Comment:
There's a problem:
if `get_size()` is less than 200, then `sample_num` is zero, which will cause divided error.

So `const size_t sample_num = this->get_size() * sample_rate + 1;` is better, I think.